### PR TITLE
SDDSIP-1726 - Implement a state machine approach for the navigation of returns screens

### PR DESCRIPTION
**Pull request by richard-mohammed-def**

### CODE REVIEW

**Comment by richard-mohammed-def** [comment]
I've tried improving the "code smell" according to SonarCloud guidelines, but I think changing the code anymore will make it difficult to read. Would like a discussion on the setting on SonarCloud as I think by having it as strict as it is we're creating a rod for our own backs and it will slow us down significantly.

File: packages/web-service/src/pages/returns/common-return-functions.js [hunk 1]
```diff
@@ -119,3 +121,18 @@ export const getWhyNoArtificialSettReason = returnData => {
   }
   return whyNoArtificialSettReason
 }
+
+export const allCompletion = async (request) => {
+  const journeyData = await request.cache().getData()
+
+  const returnId = journeyData?.returns?.id
```
Comment (GrahamMWillis) [inline packages/web-service/src/pages/returns/common-return-functions.js:128]: if .returns undefined is possible we should have a check function throw the user back to the applications page and expect .returns defined

**Review by GrahamMWillis — COMMENTED** [review]
We are not at liberty to change the rules set in Sonar -- they are dictated by the defra developer community, however I do override certain rules on occasion.

I think in this case the cognitive complexity, function length and multiple return points are appropriate for this function. I have Ok'd thoes in Sonar.

The remainder you will have to fix I'm afraid, mostly just  removing the brackets (x) =>

File: packages/web-service/src/pages/returns/common-return-functions.js (hunk 1 above)
Comment (richard-mohammed-def) [reply packages/web-service/src/pages/returns/common-return-functions.js:128]: Totally. Will add this 👍

**Review by GrahamMWillis — APPROVED** [review]

**Pull request merged**